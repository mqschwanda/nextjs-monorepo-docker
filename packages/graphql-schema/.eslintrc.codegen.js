module.exports = {
  extends:
  [
    'eslint-config-codegen',
  ],
  parserOptions: {
    project: [
      './tsconfig.eslint.json',
    ],
  },
  root: true,
};
