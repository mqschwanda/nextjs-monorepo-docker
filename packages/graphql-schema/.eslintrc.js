module.exports = {
  extends: [
    '@mqs/eslint-config/client',
  ],
  parserOptions: {
    project: [
      './tsconfig.eslint.json',
    ],
  },
  root: true,
  rules: {
    'import/no-extraneous-dependencies': [
      'error',
      {
        devDependencies: [
          './eslintrc.codegen.js',
          './eslintrc.js',
          './codegen.ts',
          '**/__tests__/**/*',
        ],
      },
    ],
  },
};
