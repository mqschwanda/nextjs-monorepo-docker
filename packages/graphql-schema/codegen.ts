import path from 'path';
import { CodegenConfig } from '@graphql-codegen/cli';
import {
  documentsGlob,
  generatedDir,
  reactApolloVersion,
  schemaGlob,
} from './src/constants';

export const srcDir = 'src';
export const typesPath = './graphql';

const mappers = {
  JobKey: '@mqs/prisma/client#JobKey',
  Log: '@mqs/prisma/client#Log',
  LogType: '@mqs/prisma/client#LogType',
  RoleKey: '@mqs/prisma/client#RoleKey',
  User: '@mqs/prisma/client#User',
};

const config: CodegenConfig = {
  documents: [
    path.join(srcDir, documentsGlob),
  ],
  generates: {
    [path.join(srcDir, generatedDir, '/')]: {
      config: {
        arrayInputCoercion: false,
        avoidOptionals: false,
        dedupeFragments: true,
        defaultScalarType: 'any',
        enumsAsTypes: true,
        mappers,
        namingConvention: 'change-case-all#pascalCase',
        nonOptionalTypename: false,
        scalars: {
          Date: 'Date',
          ID: {
            input: 'string',
            output: 'string | number',
          },
          JSON: 'JSON',
        },
        skipTypename: true,
        strictScalars: true,
        useTypeImports: false,
      },
      plugins: [

      ],
      preset: 'client',
      presetConfig: {
        fragmentMasking: false,
        typesPath,
      },
    },
    [path.join(srcDir, generatedDir, 'resolvers.ts')]: {
      config: {
        mappers,
        reactApolloVersion,
      },
      plugins: [
        'typescript-resolvers',
      ],
      preset: 'import-types',
      presetConfig: {
        typesPath,
      },
    },
  },
  hooks: {
    afterAllFileWrite: 'yarn eslint --config .eslintrc.codegen.js --fix',
  },
  schema: [
    {
      [path.join(srcDir, schemaGlob)]: {
        noRequire: true,
      },
    },
  ],
};

export default config;
