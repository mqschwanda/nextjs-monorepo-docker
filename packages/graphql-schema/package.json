{
  "name": "graphql-schema",
  "version": "0.0.0",
  "license": "MIT",
  "sideEffects": false,
  "main": "./dist/index.js",
  "source": "./src/index.ts",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": "./dist/index.js",
    "./constants": "./dist/constants.js",
    "./loadSchema": "./dist/loadSchema.js",
    "./package.json": "./package.json",
    "./queries/*.graphql": "./dist/queries/*.graphql",
    "./schema/*.graphql": "./dist/schema/*.graphql"
  },
  "typesVersions": {
    "*": {
      "constants": [
        "./dist/constants.d.ts"
      ],
      "loadSchema": [
        "./dist/loadSchema.d.ts"
      ],
      "react-apollo": [
        "./dist/__generated__/react-apollo.d.ts"
      ]
    }
  },
  "files": [
    "dist/**"
  ],
  "scripts": {
    "codegen": "graphql-codegen **/*.graphql",
    "codegen:watch": "yarn codegen --watch",
    "copy-schema": "copyfiles -u 1 \"./src/schema/**/*\" \"./dist/\"",
    "copy-schema:watch": "nodemon --watch ./src/schema/* --exec \"yarn copy-schema\"",
    "copy-queries": "copyfiles -u 1 \"./src/queries/**/*\" \"./dist/\"",
    "copy-queries:watch": "nodemon --watch ./src/queries/* --exec \"yarn copy-queries\"",
    "build": "yarn codegen && yarn build-tsc && yarn copy-schema && yarn copy-queries",
    "build-tsc": "tsc",
    "build-tsc:watch": "yarn build -w",
    "clean": "rm -rf dist",
    "dev": "yarn build && concurrently \"yarn codegen:watch\" \"yarn build-tsc:watch\" \"yarn copy-schema:watch\" \"yarn copy-queries:watch\"",
    "lint": "eslint .",
    "lint:fix": "yarn lint --fix"
  },
  "devDependencies": {
    "@graphql-codegen/cli": "^3.3.1",
    "@graphql-codegen/client-preset": "^3.0.1",
    "@graphql-codegen/import-types-preset": "^2.2.6",
    "@graphql-codegen/typed-document-node": "^4.0.1",
    "@graphql-codegen/typescript-resolvers": "^3.2.1",
    "@graphql-eslint/eslint-plugin": "^3.18.0",
    "concurrently": "^8.0.1",
    "copyfiles": "^2.4.1",
    "eslint": "^7.32.0",
    "eslint-config-codegen": "*",
    "eslint-config-custom": "*",
    "nodemon": "^2.0.22",
    "react": "^18.2.0",
    "tsconfig": "*",
    "typescript": "^4.5.2"
  },
  "dependencies": {
    "@apollo/client": "^3.7.13",
    "glob": "^10.2.3",
    "graphql": "^16.6.0"
  }
}
