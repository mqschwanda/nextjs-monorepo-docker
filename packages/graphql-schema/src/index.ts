import { buildSchema } from 'graphql';

// Construct a schema, using GraphQL schema language
const schema = buildSchema(`
  type Query {
    hello: String
  }
`);

export default schema;
