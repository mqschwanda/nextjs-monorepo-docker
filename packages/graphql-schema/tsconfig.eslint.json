{
  "extends": "./tsconfig.json",
  "files": [
    "./codegen.ts",
    "./.eslintrc.js",
    "./.eslintrc.codegen.js"
  ],
}
