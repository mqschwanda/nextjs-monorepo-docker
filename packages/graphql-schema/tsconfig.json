{
  "extends": "@mqs/tsconfig/base.json",
  "include": [
    "./src"
  ],
  "exclude": [
    "dist",
    "build", 
    "node_modules"
  ],
  "compilerOptions": {
    "baseUrl": "./src",
    "lib": [
      "ES2015"
    ],
    "module": "CommonJS",
    "outDir": "./dist",
    "rootDir": "./src",
    "paths": {}
  },
}
