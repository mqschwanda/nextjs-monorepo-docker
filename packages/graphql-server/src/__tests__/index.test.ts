import { buildHTTPExecutor } from '@graphql-tools/executor-http';
import { CountdownDocument, HelloDocument } from '@mqs/graphql-schema';
import { isAsyncIterable } from 'graphql-yoga';
import createGraphqlServer from '..';

describe('@mqs/graphql-server', () => {
  describe('Query', () => {
    describe('hello', () => {
      it('should return data', async () => {
        const name = 'John Smith';

        const graphqlServer = createGraphqlServer();
        const executor = buildHTTPExecutor({
          endpoint: graphqlServer.graphqlEndpoint,
          fetch: graphqlServer.fetch,
        });
        const result = await executor({
          document: HelloDocument,
          variables: {
            name,
          },
        });

        if (isAsyncIterable(result)) {
          throw new Error('result should not be AsyncIterable');
        }

        expect(result).toMatchSnapshot();
        expect(result.data?.hello).toBe(`Hello ${name}`);
      });
    });
  });
  describe('Subscription', () => {
    describe('countdown', () => {
      it('should return async iterable data', async () => {
        const from = 2;

        const graphqlServer = createGraphqlServer();
        const executor = buildHTTPExecutor({
          endpoint: graphqlServer.graphqlEndpoint,
          fetch: graphqlServer.fetch,
        });
        const subscription = await executor({
          document: CountdownDocument,
          variables: {
            from,
          },
        });

        if (!isAsyncIterable(subscription)) {
          throw new Error('result should be AsyncIterable');
        }

        let i = from;
        // eslint-disable-next-line no-restricted-syntax
        for await (const result of subscription) {
          expect(result.data?.countdown).toBe(i);
          i--;

          if (i === 0) {
            break;
          }
        }
      });
    });
  });
});
