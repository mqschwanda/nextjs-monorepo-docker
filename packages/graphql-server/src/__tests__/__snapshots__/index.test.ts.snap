// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/graphql-server Query hello should return data 1`] = `
Object {
  "data": Object {
    "hello": "Hello John Smith",
  },
}
`;
